src/debugger/Debugger.js

```diff
@@ -97,56 +95,73 @@ export class DebugServer {
       if (filePath === null) return;
       let lineNum = location.start.line;
       let colNum = location.start.column;
-
       // Check whether there is a breakpoint we need to stop on here
-      if (!this.proceedBreakpoint(filePath, lineNum, colNum)) return;
-
+      let breakpoint = this.findStoppableBreakpoint(filePath, lineNum, colNum);
+      if (breakpoint === null) return;
       // Tell the adapter that Prepack has stopped on this breakpoint
-      this.channel.writeOut(`breakpoint stopped ${lineNum}:${colNum}`);
+      this.channel.writeOut(
+        `${DebugMessage.BREAKPOINT_STOPPED_RESPONSE} ${breakpoint.filePath} ${breakpoint.line}:${breakpoint.column}`
+      );
 
       // Wait for the adapter to tell us to run again
-      this.waitForRun(function(line) {
-        return line === "proceed";
-      });
+      this.waitForRun();
     }
   }
 
-  executeCommand(command: string) {
+  // Process a command from a debugger. Returns whether Prepack should unblock
+  // if it is blocked
+  processDebuggerCommand(command: string) {
     if (command.length === 0) {
       return;
     }
     let parts = command.split(" ");
-    if (parts[0] === "breakpoint") {
-      this.executeBreakpointCommand(this._parseBreakpointArguments(parts));
-    }
-  }
-
-  executeBreakpointCommand(args: BreakpointCommandArguments) {
-    if (args.kind === "add") {
-      this.breakpoints.addBreakpoint(args.filePath, args.lineNum, args.columnNum);
-      this.channel.writeOut(`added breakpoint ${args.filePath} ${args.lineNum} ${args.columnNum}`);
-    } else if (args.kind === "remove") {
-      this.breakpoints.removeBreakpoint(args.filePath, args.lineNum, args.columnNum);
-      this.channel.writeOut(`removed breakpoint ${args.filePath} ${args.lineNum} ${args.columnNum}`);
-    } else if (args.kind === "enable") {
-      this.breakpoints.enableBreakpoint(args.filePath, args.lineNum, args.columnNum);
-      this.channel.writeOut(`enabled breakpoint ${args.filePath} ${args.lineNum} ${args.columnNum}`);
-    } else if (args.kind === "disable") {
-      this.breakpoints.disableBreakpoint(args.filePath, args.lineNum, args.columnNum);
-      this.channel.writeOut(`disabled breakpoint ${args.filePath} ${args.lineNum} ${args.columnNum}`);
+    let prefix = parts[0];
+    switch (prefix) {
+      case DebugMessage.BREAKPOINT_ADD_COMMAND:
+        let addArgs = this._parseBreakpointArguments(parts);
+        this.breakpoints.addBreakpoint(addArgs.filePath, addArgs.lineNum, addArgs.columnNum);
+        this.channel.writeOut(
```

Comment:
this.channel.writeOut() contains the details of the format of the message which may change. Please take out and hide all the "this.channel.writeOut" calls behind one helper function. Then one day we want to change the format, only one place needs to be changed instead of every switch case.